src/m365/teams/commands/message/message-export.ts

```diff
@@ -0,0 +1,218 @@
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        teamId: typeof args.options.teamId !== 'undefined',
+        fromDateTime: typeof args.options.fromDateTime !== 'undefined',
+        toDateTime: typeof args.options.toDateTime !== 'undefined',
+        withAttachments: !!args.options.withAttachments,
+        folderPath: typeof args.options.folderPath !== 'undefined'
+      });
+    });
+  }
+
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userName [userName]'
+      },
+      {
+        option: '--teamId [teamId]'
+      },
+      {
+        option: '--fromDateTime [fromDateTime]'
+      },
+      {
+        option: '--toDateTime [toDateTime]'
+      },
+      {
+        option: '--withAttachments'
+      },
+      {
+        option: '--folderPath [folderPath]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.userName && !validation.isValidUserPrincipalName(args.options.userName)) {
+          return `${args.options.userName} is not a valid userPrincipalName`;
+        }
+
+        if (args.options.teamId && !validation.isValidGuid(args.options.teamId)) {
+          return `${args.options.teamId} is not a valid GUID`;
+        }
+
+        if (args.options.fromDateTime && !validation.isValidISODateTime(args.options.fromDateTime)) {
+          return `${args.options.fromDateTime} is not a valid ISO DateTime`;
+        }
+
+        if (args.options.toDateTime && !validation.isValidISODateTime(args.options.toDateTime)) {
+          return `${args.options.toDateTime} is not a valid ISO DateTime`;
+        }
+
+        if (args.options.folderPath && !fs.existsSync(args.options.folderPath)) {
+          return `Path ${args.options.folderPath} does not exist.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      {
+        options: ['userId', 'userName', 'teamId']
+      },
+      {
+        options: ['folderPath'],
+        runsWhen: (args: CommandArgs) => {
+          return args.options.withAttachments !== undefined && args.options.withAttachments;
+        }
+      },
+      {
+        options: ['withAttachments'],
+        runsWhen: (args: CommandArgs) => {
+          return args.options.folderPath !== undefined;
+        }
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let baseUrl = `${this.resource}/v1.0/`;
```

Comment:
This until the line where you assign the variable requestUrl is quite verbose. Why not write it in a single terniary statement?

```ts
const requestUrl = args.options.userId || args.options.userName
    ? `${this.resource}/v1.0/users/${args.options.userId || args.options.userName}/chats/getAllMessages`
    : `${this.resource}/v1.0/teams/${args.options.teamId}/channels/getAllMessages`;
```